src/filter_tree/filter_tree.c

```diff
@@ -418,6 +418,155 @@ void FilterTree_DeMorgan(FT_FilterNode **root) {
 	_FilterTree_DeMorgan(root, 0);
 }
 
+// This method returns if an arithmetic expression is valid to be use in filter tree compaction.
+static inline bool _FT_FilterNode_constnat_expression(AR_ExpNode *exp) {
+	// TODO: Handle injected parames.
+	return exp->type == AR_EXP_OPERAND && exp->operand.type == AR_EXP_CONSTANT;
+}
+
+// Compacts an expression node. Return if this node can be used in compression.
+static inline bool _FilterTree_Compact_Exp(FT_FilterNode *node) {
+	return _FT_FilterNode_constnat_expression(node->exp.exp);
+}
+
+// In place set an existing filter tree node to expression node.
+static inline void _FilterTree_In_Place_Set_Exp(FT_FilterNode *node, SIValue v) {
+	node->t = FT_N_EXP;
+	node->exp.exp = AR_EXP_NewConstOperandNode(v);
+}
+
+// Compacts 'AND' condition node.
+static bool _FilterTree_Compact_And(FT_FilterNode *node) {
```

Comment:
I think it would be clearer if after determining both `is_lhs_const` and `is_rhs_const`
we would:
```
if(!is_lhs_const && !is_rhs_const) {
return false;
} elseif(is_lhs_const && is_rhs_const) {
// both...
...
} elseif(is_lhs_const) {
// left hand side,
...
} else {
// right hand side.
...
}
```

Reply:
done